### PR TITLE
haku: mixed resolution training

### PR DESCRIPTION
**Pull request by morganmcg1**
# Hypothesis: Mixed-Resolution Training for Surface and Volume Points

## Motivation

The current SOTA uses a fixed 65536 surface + 65536 volume points for both training and evaluation. However, the model likely learns different feature statistics depending on point density — high-density sampling captures fine boundary-layer gradients near sharp edges and curvature discontinuities, while low-density sampling forces the model to rely on longer-range context.

Mixed-resolution training (randomly varying the number of surface/volume points each batch) acts as a form of implicit regularization: the model cannot overfit to any single point density and must learn density-invariant representations. This is related to DropPoint, random masking, and multi-scale training techniques that have improved robustness in point cloud processing literature.

For τ_y/τ_z specifically, the model currently only sees 65536 surface points — the 3D wake regions that drive crossflow shear may be undersampled. Random upsampling to 131072 points in some batches could help.

## Hypothesis

Training with randomly varied surface/volume point counts (uniformly sampled from {32768, 65536, 131072} each batch) will improve generalization and reduce τ_y/τ_z gaps by forcing the model to learn density-invariant features.

## Current Best Metrics (Merge Bar)

| Metric | val | test |
|---|---|---|
| abupt_axis_mean_rel_l2_pct | **7.3914%** | 8.7189% |
| τ_y (wallshear_y) | ~9.61% | ~11.0% |
| τ_z (wallshear_z) | ~11.06% | ~12.7% |

AB-UPT reference: sp=3.82%, ws=7.29%, vp=6.08%, τ_x=5.35%, τ_y=3.65%, τ_z=3.63%

## Implementation Instructions

The simplest approach: modify the data loader to randomly sample a number of points per batch from a discrete set, rather than a fixed count. Check if a `--train-surface-points-range` or `--mixed-resolution` flag exists in train.py:

```bash
python train.py --help | grep -i resol
python train.py --help | grep -i points
```

If a mixed-resolution flag exists, use it. Otherwise, the simplest code modification is to randomly choose `n_surface` from `{32768, 65536, 131072}` at the start of each training batch.

**Arm A — Fixed 131072 surface points (max density, baseline comparison):**
```bash
torchrun --standalone --nproc_per_node=1 train.py \
  --agent kohaku --wandb-group haku-mixed-resolution-training \
  --learnable-pe --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --clip-grad-norm 0.5 --grad-ema-alpha 0.5 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.999 --batch-size 4 --validation-every 1 --no-compile-model \
  --train-surface-points 131072 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --surface-curvature-features k1_k2
```

**Arm B — Mixed resolution (if flag exists) or code-level modification:**
If `--mixed-resolution` or similar flag exists, use it with surface points ranging over {32768, 65536, 131072}. Otherwise implement it in the data loader and run:
```bash
torchrun --standalone --nproc_per_node=1 train.py \
  --agent kohaku --wandb-group haku-mixed-resolution-training \
  --learnable-pe --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --clip-grad-norm 0.5 --grad-ema-alpha 0.5 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --ema-decay 0.999 --batch-size 4 --validation-every 1 --no-compile-model \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --surface-curvature-features k1_k2
```

Run Arm A first as it requires no code change. If Arm A is already in the repo (check if a PR for 131072 surface points exists), run Arm B instead with the code modification.

## Success Criteria

- `abupt_axis_mean_rel_l2_pct` < **7.3914%** on validation is the merge bar.
- Report per-axis metrics — particularly τ_y and τ_z to see if higher density helps crossflow components.

## Terminal Result Format

```
SENPAI-RESULT: {"terminal":true,"status":"complete","pending_arms":false,"wandb_run_ids":["<run-id-a>","<run-id-b>"],"primary_metric":{"name":"val/abupt_axis_mean_rel_l2_pct","value":<val>},"test_metric":{"name":"test/abupt_axis_mean_rel_l2_pct","value":<test>}}
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #706 because ADVISOR: Closing orphaned draft. kohaku currently has active WIP PR #719 (SDF-stratified volume norm) — duplicate assignment from earlier round superseded by current work..
